code/datums/components/crafting/crafting.dm

```diff
@@ -451,6 +451,38 @@
 	data["display_compact"] = display_compact
 
 	var/list/surroundings = get_surroundings(user)
+	var/fueled_welder_found = FALSE
+
+	// set as found as these omnitools don't have fuel, they just work
+	if ((/obj/item/debug/omnitool in surroundings["instances"]) || (/obj/item/abductor/alien_omnitool in surroundings["instances"]))
+		fueled_welder_found = TRUE
```

Comment:
Yeah this isn't super maintainable in this form. We don't want to necessitate people come here and update this every time they add a new welding item.

I was reminded that there exists a proc which determines if a tool is usable, `tool_use_check`
You should just be able to utilize it to determine if the welter, or plasma cutter, or etc. is usable, ("has fuel")

Something like:

```dm
for(var/obj/item/thing as anything in surroundings["instances"])
    if(thing.tool_behavior == TOOL_WELDER && thing.tool_use_check(user, 1))
        num_usable_welders++
```

However, as I'm writing this i'm also realizing, why don't we just *prevent* unusable welders from being considered as usable? 

We can just modify `get_surroundings()`:
```dm
		else //a reagent container that is empty can also be used as a tool. e.g. glass bottle can be used as a rolling pin
			if(item.tool_behaviour)
				.["tool_behaviour"] += item.tool_behaviour
```
to

```dm
		else //a reagent container that is empty can also be used as a tool. e.g. glass bottle can be used as a rolling pin
			if(item.tool_behaviour && item.tool_use_check(a, 1))
				.["tool_behaviour"] += item.tool_behaviour
```

Reply:
Oh minor hitch, `tool_use_check` is impure, annoying. So it'd spam people's chat. You might need to modify it slightly to prevent that.

Reply:
I see, yeah this could work better you are right. Could you explain what makes the function impure though? I am not quite sure after looking at it. Also why does it being impure make it spam people's chat? I would be happy to change it if you could explain these two things.

Reply:
The proc will send chat messages to the user saying "you can't use that, not enough fuel". It'll spam people because `get_surroundings` gets called like 5 times a second for `ui_data`. 

You could easily just add a proc argument, `silent = FALSE`. Don't send chat messages if silent is true, pass silent as true for get surroundings

Reply:
Oh I see what I did wrong, I looked at the base proc `tool_use_check` of items rather than the one for welding tools and the like, that's why I was confused. Yeah I can do it easily.

Reply:
The problem with the `tool_use_check` is that for everything that is not a welder/igniter/plasmacutter with fuel, it just returns false, because the default function is `return !amount`. I am not sure if there is a way to check if the `tool_use_check` function has been overwritten to only work for these tools but it is a problem for now.

Example:

`if(item.tool_behaviour && item.tool_use_check(a, 1, silent = TRUE))`

this returns `FALSE` for a screwdriver to be used as tool, while it returns `TRUE` for a welding tool because of the implemented functionality.

What do you think is the appropriate response to this problem? Seems like `tool_use_check` is used in quite a lot of places, I am not knowledgable enough on the codebase to know what change would be ideal.

Reply:
We can get around it by only passing `amount = 1` if it's a welder, I think that's the intended way to use it... Or it's just jank
`item.tool_use_check(a, amount = (item.tool_behaviour == TOOL_WELDER ? 1 : 0), silent = TRUE)`

Reply:
It is a good idea, but the alien omnitool and the debug omnitool do not have their own `tool_use_check` functions and therefore can't be used for crafting. Also I think with this code change it forces the welding tool to be open for it to be passed as working. This small fix is becoming more janky by the second...